src/internal/base-component/telemetry.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect } from 'react';
+import { Metrics } from '../metrics/metrics';
+
+export function useTelemetry(metrics: Metrics, componentName: string, theme: string) {
```

Comment:
Why is this injected? In all use-cases, we use the same metrics provider, never changes dynamicaly

Reply:
The metrics object is instantiated with package source and package version. If the metrics will be instantiated here the function will no longer be suitable to move to the toolkit.

Reply:
Discussed in DM, proposed solution is to provide all variables via arguments

```
function useTelemetry(package: string, version: string, componentName: string, theme: string)
```